community/kernel/src/main/java/org/neo4j/kernel/impl/storageengine/impl/recordstorage/StoreStatement.java

```diff
@@ -331,123 +338,119 @@ public Properties properties()
+            relationshipStore.getRecordByCursor( reference, relationshipRecord, mode, cursor );
         }
 
         @Override
         public long getHighestPossibleIdInUse()
         {
-            return neoStores.getRelationshipStore().getHighestPossibleIdInUse();
+            return relationshipStore.getHighestPossibleIdInUse();
         }
     }
 
     class Groups implements StorageStatement.Groups
     {
-
         @Override
         public PageCursor openPageCursor( long reference )
         {
-            return neoStores.getRelationshipGroupStore().openPageCursorForReading( reference );
+            return relationshipGroupStore.openPageCursorForReading( reference );
         }
 
         @Override
         public void loadRecordByCursor( long reference, RelationshipGroupRecord relationshipGroupRecord,
                 RecordLoad mode, PageCursor cursor ) throws InvalidRecordException
         {
-            neoStores.getRelationshipGroupStore().getRecordByCursor( reference, relationshipGroupRecord, mode, cursor );
+            relationshipGroupStore.getRecordByCursor( reference, relationshipGroupRecord, mode, cursor );
         }
 
         @Override
         public long getHighestPossibleIdInUse()
         {
-            return neoStores.getRelationshipGroupStore().getHighestPossibleIdInUse();
+            return relationshipGroupStore.getHighestPossibleIdInUse();
         }
     }
 
     class Properties implements StorageStatement.Properties
     {
-
         @Override
         public PageCursor openPageCursor( long reference )
         {
-            return neoStores.getPropertyStore().openPageCursorForReading( reference );
+            return propertyStore.openPageCursorForReading( reference );
         }
 
         @Override
         public void loadRecordByCursor( long reference, PropertyRecord propertyBlocks, RecordLoad mode,
                 PageCursor cursor ) throws InvalidRecordException
         {
-            neoStores.getPropertyStore().getRecordByCursor( reference, propertyBlocks, mode, cursor );
+            propertyStore.getRecordByCursor( reference, propertyBlocks, mode, cursor );
         }
 
         @Override
         public long getHighestPossibleIdInUse()
         {
-            return neoStores.getPropertyStore().getHighestPossibleIdInUse();
+            return propertyStore.getHighestPossibleIdInUse();
         }
 
         @Override
         public PageCursor openStringPageCursor( long reference )
         {
-            return neoStores.getPropertyStore().getStringStore().openPageCursorForReading( reference );
+            return propertyStore.getStringStore().openPageCursorForReading( reference );
         }
 
         @Override
         public PageCursor openArrayPageCursor( long reference )
         {
-            return neoStores.getPropertyStore().getArrayStore().openPageCursorForReading( reference );
+            return propertyStore.getArrayStore().openPageCursorForReading( reference );
```

Comment:
It'd sort of make sense to keep `propertyStore.getArrayStore()` as a field too.

Reply:
I but the array and string stores in fields because

1) it's accessed less often, and when it is accessed the access will be costly, so the relative cost of getting the store will be small.
2) They are just access methods to final fields and should be extremely cheap. The `getNodeStore()` seems to be harder for the compiler to optimize away, don't ask me why. 

Do you think I should anyway?

Reply:
I think you should, if nothing else then at least for consistency?

Reply:
👍